doc/source/io.rst

```diff
@@ -468,6 +469,18 @@ Individual columns can be parsed as a ``Categorical`` using a dict specification
 
    pd.read_csv(StringIO(data), dtype={'col1': 'category'}).dtypes
 
+Specifying ``dtype='cateogry'`` will result in a ``Categorical`` that is
+unordered, and whose ``categories`` are the unique values observed in the data.
+For more control on the categories and order, create a
+:class:`~pandas.api.types.CategoricalDtype` ahead of time.
+
+.. ipython:: python
+
+   from pandas.api.types import CategoricalDtype
+
+   dtype = CategoricalDtype(['d', 'c', 'b', 'a'], ordered=True)
+   pd.read_csv(StringIO(data), dtype={'col1': dtype}).dtypes
+
 .. note::
 
    The resulting categories will always be parsed as strings (object dtype).
```

Comment:
How clever do we want to be here? If we have `CategoricalDtype([1, 2, 3])` and see a CSV with `1,2,3`, should we interpret those as integers? I'm not sure.

Reply:
I think ideally we would - I actually had it working that way at one point the original PR, but the implementation was too complex / duplicative - so we decided not to.  But I don't think it will be as bad in the categories known in advance case.

Reply:
OK. I'm thinking we limit it to the case where all the categories are the same type. I'll see how difficult it is.